airflow/models/__init__.py

```diff
@@ -3419,9 +3449,14 @@ def get_next_run_date(self, after_date):
             next_date = after_date + schedule_interval - (after_date - start_date) % schedule_interval
             """
             start_date = min(t.start_date for t in self.tasks)
-            return after_date \
-                   - timedelta(seconds=((after_date - start_date).total_seconds() % self._schedule_interval.total_seconds())) \
-                   + self._schedule_interval
+            try:
+                intervals = json.loads(self._schedule_interval)
+                following_schedules = self.following_schedule(after_date)
+                return sorted(following_schedules)[0]
```

Comment:
`min(foo)` is better than `sorted(foo)[0]`:

```suggestion
                return min(following_schedules)
```